src/system/connectivity/wifi.cpp

```diff
@@ -15,13 +48,29 @@ static WiFiManager wm;
  * @param wifi_password  [Optional] WiFi Password as a const char string.
  * @return wl_status_t   WiFi status code.
  */
-wl_status_t WifiInitialize (
+wl_status_t WiFiConnection::WifiInitialize (
     const char* wifi_ssid,
     const char* wifi_password
     )
 {
     logger << LOG_INFO << "Initializing WiFi provider!" << EndLine;
 
+    WiFiManagerParameter templateID("template_id", "TEMPLATE_ID", "template_id",WIFI_MANAGER_PARAMETER_SIZE);
+    WiFiManagerParameter deviceName("device_name", "DEVICE_NAME", "Device_Name", WIFI_MANAGER_PARAMETER_SIZE);
+    WiFiManagerParameter authToken("auth_token", "AUTH_TOKEN", "auth_token", WIFI_MANAGER_PARAMETER_SIZE);
+
+    wm.addParameter(&templateID);
+    wm.addParameter(&deviceName);
+    wm.addParameter(&authToken);
+
+    wm.setPreSaveConfigCallback(setPreSaveConfigCallback);
+
+    wm.setTitle("Irrigation System");
+    const char *menu[] = {"wifi","info","param","sep","restart","exit"};
+    wm.setMenu(menu,6);
```

Comment:
How is memory managed by `setMenu` function?
Because the `menu` memory will be unreferenced at the end of the function, so the strings will be a memory leak.

Set menu copies the strings to it internal holder? or is taking only the string address?

Reply:
Is this memory safe?

Reply:
it is safe, it just take the values passes to compare with known options inside the library